### PR TITLE
Fixes done at 5.2.z

### PR DESCRIPTION
**Pull request by emreyigit**
Fixes various issues(mostly server version based things) that are discovered during compatibility test. Also, reversed the changes on jet enabled cluster config since jet config is added to config schema after 4.0 which causes trouble during compatibility tests.

_Note: Fixes should be ported to `master`._

### CODE REVIEW

File: src/Hazelcast.Net.Tests/Linq/LINQTests.cs [hunk 1]
```diff
@@ -29,9 +29,11 @@ namespace Hazelcast.Tests.Linq
     /// Tests the IHMap using LINQ APIs. It focuses the results mainly.
     /// For query tests, please see other linq test suites.
     /// </summary>
-    [ServerCondition("[5.0,)")] // only on server 5.0 and above
+    [ServerCondition("[5.2,)")] // only on server 5.0 and above
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net.Tests/Linq/LINQTests.cs:32]:
```suggestion
    [ServerCondition("[5.2,)")] // only on server 5.2 and above
```

File: src/Hazelcast.Net.Testing/RemoteControllerClientExtensions.cs [hunk 2]
```diff
@@ -235,8 +235,13 @@ public static async Task<(NuGetVersion, NuGetVersion)> DetectServerVersionAsync(
             if (result == null) return default;
             var resultString = Encoding.UTF8.GetString(result);
             var resultValues = resultString.Split(',');
-            var ossVersion = resultValues.Length > 0 ? NuGetVersion.Parse(resultValues[0]) : null;
-            var enterpriseVersion = resultValues.Length > 1 ? NuGetVersion.Parse(resultValues[1]) : null;
+
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net.Testing/RemoteControllerClientExtensions.cs:238]: doing it differently in `master` can you check?

**Review by zpqrtbnk — COMMENTED** [review]
commented

**Comment by emreyigit** [comment]
Closing it since https://github.com/hazelcast/hazelcast-csharp-client/pull/769 covers the current fixes.

**Pull request closed without merging**